test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -201,17 +201,26 @@ func (g *Grabber) GrabFromScheduler() (SchedulerMetrics, error) {
 	}
 
 	var err error
-	var output string
+
 	g.waitForSchedulerReadyOnce.Do(func() {
-		var lastMetricsFetchErr error
-		if metricsWaitErr := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
-			output, lastMetricsFetchErr = g.getSecureMetricsFromPod(g.kubeScheduler, metav1.NamespaceSystem, kubeSchedulerPort)
-			return lastMetricsFetchErr == nil, nil
-		}); metricsWaitErr != nil {
-			err = fmt.Errorf("error waiting for scheduler pod to expose metrics: %v; %v", metricsWaitErr, lastMetricsFetchErr)
-			return
+		if readyErr := e2epod.WaitForPodsReady(g.client, metav1.NamespaceSystem, g.kubeScheduler, 0); readyErr != nil {
+			err = fmt.Errorf("error waiting for kube-scheduler pod to be ready: %w", readyErr)
 		}
 	})
+	if err != nil {
+		return SchedulerMetrics{}, err
+	}
+
+	var lastMetricsFetchErr error
+	var output string
+	if metricsWaitErr := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		output, lastMetricsFetchErr = g.getSecureMetricsFromPod(g.kubeScheduler, metav1.NamespaceSystem, kubeSchedulerPort)
+		return lastMetricsFetchErr == nil, nil
+	}); metricsWaitErr != nil {
+		err := fmt.Errorf("error waiting for kube-scheduler pod to expose metrics: %v; %v", metricsWaitErr, lastMetricsFetchErr)
+		return SchedulerMetrics{}, err
+	}
+
 	if err != nil {
```

Comment:
this `if err != nil ` is not needed now, it's checked in L210
https://github.com/kubernetes/kubernetes/pull/103806/files#diff-28a3131f96324063dd53e17270d435a3b0b3bd8f806ee0e33295929570eab209R210

Reply:
you return if metricsWaitErr != nil

Reply:
thanks for catching this, I've updated all of these places :)